### PR TITLE
Add build, release and PR check pipelines

### PR DESCRIPTION
**Pull request by nilushancosta**
### Description
Add build, release and PR check pipelines

### CODE REVIEW

**Review by tharinduwijewardane — APPROVED** [review]

File: .azure/wso2-opensearch-k8s-operator-release.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    repository: $(REPOSITORY)
+    tags: $(APP_REVISION)
+- template: install/install-trivy.yaml@common-templates
+- template: pr/trivy-scan.yaml@common-templates
+  parameters:
+    IMAGES:
+    - $(CONTAINER_REGISTRY)/$(REPOSITORY):$(APP_REVISION)
+- script: |
+    docker tag $(CONTAINER_REGISTRY)/$(REPOSITORY):$(APP_REVISION) $(PRIVATE_CONTAINER_REGISTRY)/$(REPOSITORY):$(APP_REVISION)
+    docker tag $(CONTAINER_REGISTRY)/$(REPOSITORY):latest $(PRIVATE_CONTAINER_REGISTRY)/$(REPOSITORY):latest
```
Comment (SachithKasthuriarachchi) [inline .azure/wso2-opensearch-k8s-operator-release.yaml]: there is no docker build for latest tag it seems

Comment (nilushancosta) [reply .azure/wso2-opensearch-k8s-operator-release.yaml]: Fixed through https://github.com/wso2/wso2-opensearch-k8s-operator/pull/2/commits/c901a4aceaa9c4fa05f3ed9e0217b632301ddfb5

File: .azure/wso2-opensearch-k8s-operator-release.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    tags: $(APP_REVISION)
+- task: Docker@2
+  displayName: "Push Docker image to Choreo Private ACR"
+  inputs:
+    command: push
+    containerRegistry: "choreo-private-acr"
+    repository: $(REPOSITORY)
+    tags: |
+      $(APP_REVISION)
+      latest
```
Comment (SachithKasthuriarachchi) [inline .azure/wso2-opensearch-k8s-operator-release.yaml]:
```suggestion
```

Comment (nilushancosta) [reply .azure/wso2-opensearch-k8s-operator-release.yaml]: Fixed through https://github.com/wso2/wso2-opensearch-k8s-operator/pull/2/commits/22812efbce3c3a04a85584ffd71590b192bf1d0b

**Pull request merged**